CIP-XXXX/README.md

```diff
@@ -0,0 +1,46 @@
+---
+CIP: ?
+Title: CIP-30 ext: Account public key
+Category: Wallets
+Status: Proposed
+Authors:
+    - Ola Ahlman <ola.ahlman@tastenkunst.io>
+    - Andrew Westberg <andrewwestberg@gmail.com>
+Implementors: [Eternl, newm-chain]
+Discussions:
+    - https://github.com/cardano-foundation/cips/pulls/?
+Created: 2023-09-03
+License: CC-BY-4.0
+---
+
+## Abstract
+This CIP extends [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/) to provide an additional endpoint for dApp to get the extended account public key from a connected wallet.
+
+## Motivation: why is this CIP necessary?
+Normally it's up to the wallet to handle the logic for utxo selection, derived addresses etc through the established CIP-30 api. Sometimes however, dApp needs greater control due to subpar utxo selection or other specific needs that can only be handled by chain lookup from derived address(es). This moves the control and complexity from wallet to dApp for those dApps that prefer this setup. A dApp has better control and can make a more uniform user experience. By exporting only the account public key, this gives read-only access to the dApp.
+
+## Specification
+A new endpoint is added namespaced according to this cip extension number that returns the connected account extended public key as [cbor\<T>](https://cips.cardano.org/cips/cip30/#cbort) defined in CIP30.
+
+### 1. api.cip????.getAccountPub(): Promise\<cbor\<Bip32PublicKey>>
+
+Errors: APIError
+
+Returns hex-encoded string representing cbor of extended account public key. Throws APIError if needed as defined by CIP30.
+
+## Rationale: how does this CIP achieve its goals?
+Raw cbor is returned instead of bech32 encoding to follow specification of other CIP30 endpoints. Wallets implementing this CIP should, but not enforced, request additional access from the user to access this endpoint as it allows for complete read access to account history and derivation paths. 
```

Comment:
> this CIP should, but not enforced

Should they or not? IMO sharing public keys should not require permission.

Reply:
It's a bit subtle, but connecting a wallet tells the dApp about all the currently used utxos and addresses of a wallet. Sharing the account public keys tells the dApp about any currently used utxos and addresses along with PAST/FUTURE addresses.  It's basically asking for an updatable read-only copy of the user's wallet instead of just the current snapshot of a user's wallet.

Reply:
I think wallets SHOULD request additional access for the reason Andrew mentioned. But I wasn't sure if CIP should mention it as enforced or just as a recommendation. This is why I expressed myself as I did. In addition to past/future addresses, it will also allow you to derive not just payment and stake keys but also upcoming DRep pub keys (CIP-95).